daemon/cmd/state.go

```diff
@@ -342,6 +342,11 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState, endpoi
 		}
 		log.WithField(logfields.EndpointID, ep.ID).Info("Successfully restored endpoint. Scheduling regeneration")
 		go func(ep *endpoint.Endpoint, epRegenerated chan<- bool) {
+
+			// Wait for the ipcache to write at least once before regenerating.
+			// Otherwise, there could be transient policy drops.
+			d.ipcache.WaitForFirstLabelInjection()
```

Comment:
This case makes sense.

Does the base programs generation need to likewise block? Or is there already some ordering constraint there somehow, for instance because the daemon does the `initMaps()` strictly before ipcache starts based on the current ordering, and therefore the new ipcache map is already created when the ipcache starts generating entries? I think that's the case.

As far as I understand this will also indirectly block each endpoint regeneration on syncing all k8s state. I think this is probably the correct behaviour but may surprise folks in big clusters.